components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.probes/src/main/java/org/wso2/micro/integrator/probes/ReadinessResource.java

```diff
@@ -87,15 +88,17 @@ public boolean invoke(MessageContext synCtx) {
                 String faultyList = String.join("\",\"", faultyCapps);
                 response += "\"status\": \"not ready, faulty CAPPs detected\", \"Faulty CAPPs\" : [\"" + faultyList +
                         "\"]}";
-                axisCtx.setProperty(HTTP_SC, 500);
+                CACHED_RESPONSE_CODE = 500;
             } else {
                 response += "\"status\" : \"ready\"}";
-                axisCtx.setProperty(HTTP_SC, 200);
+                CACHED_RESPONSE_CODE = 200;
             }
             // update cache
             CACHED_RESPONSE = response;
         }
 
+        axisCtx.setProperty(HTTP_SC, CACHED_RESPONSE_CODE);
+
         try {
             JsonUtil.getNewJsonPayload(axisCtx, response, true, true);
```

Comment:
We can use the CACHED_RESPONSE just as the 'CACHED_RESPONSE_**CODE**' for better clarity.